tests/e2e_tests/integration/07-saml.spec.ts

```diff
@@ -0,0 +1,183 @@
+import axios from 'axios';
+import * as fs from 'fs';
+import * as https from 'https';
+import jwtDecode from 'jwt-decode';
+
+import test, { expect } from '../fixtures/fixtures';
+
+test.describe('SAML Login', () => {
+  test.use({ storageState: 'storage.json' });
+  test.describe('SAML login via sso/id/login', () => {
+    test('Logs in using UI', async ({ environment, context, baseUrl, page, password, username }) => {
+      test.skip(environment !== 'staging');
+      test.setTimeout(240000);
+      const saml_settings={
+        'login':'morty',
+        'password':'panic',
+        'email':'msmith@samltest.id',
+        'idp_url':'https://samltest.id/saml/idp'
+      };
+      const timeout_4s=4000;
+      const timeout_1s=1000;
+      const storage=await context.storageState();
+      const jwt=storage['cookies'].find(cookie => cookie.name === 'JWT').value;
+
+      const idp_response = await axios({
+        url: saml_settings['idp_url'],
+        method: 'GET'
+      });
+
+      if (idp_response.status !== 200) {
+        throw 'expected 201';
+      }
+
+      const metadata = idp_response.data;
+
+      console.log(`running with ${baseUrl}`);
+      console.log(`IdP metadata len=${metadata.length} loaded and uploading`);
+
+      await page.goto(`${baseUrl}ui/settings/organization-and-billing`);
+
+      await expect(page).toHaveURL(`${baseUrl}ui/settings/organization-and-billing`);
+      // Check input[type="checkbox"]
+      await page.locator('input[type="checkbox"]').check();
+      // Click text=input with the text editor
+      await page.locator('text=input with the text editor').click();
+      // Press Tab
+      await page.locator('text=SAML metadataImport from a fileEnter to Rename, ⇧Enter to PreviewCancelSave').press('Tab');
+      // Press Tab
+      await page.locator('text=SAML metadataImport from a file >> div[role="presentation"]').press('Tab');
+      // Press Tab
+      await page.locator('text=Import from a file').press('Tab');
+      // Press Tab
+      await page.locator('text=SAML metadataImport from a file >> button').nth(1).press('Tab');
```

Comment:
This seems very brittle, whenever the layout changes in an unrelated way, these interactions will most likely break... Keep in mind: the test generation you use only generates a starting point for the test, but you should still think that all the steps are useful.

Reply:
I am thinking all the time. replaced with single click.